sonic_platform_base/sonic_xcvr/api/public/cmis.py

```diff
@@ -180,24 +180,24 @@ def get_transceiver_bulk_status(self):
         bulk_status = {
             "rx_los": all(rx_los.values()) if self.get_rx_los_support() else 'N/A',
             "tx_fault": all(tx_fault.values()) if self.get_tx_fault_support() else 'N/A',
-            "tx_disable": all(tx_disable),
+            "tx_disable": all(tx_disable) if self.get_tx_disable_support() else 'N/A',
             "tx_disabled_channel": tx_disabled_channel,
             "temperature": temp,
             "voltage": voltage
         }
 
         for i in range(1, self.NUM_CHANNELS + 1):
-            bulk_status["tx%dbias" % i] = tx_bias['LaserBiasTx%dField' % i]
-            bulk_status["rx%dpower" % i] = rx_power['OpticalPowerRx%dField' %i]
-            bulk_status["tx%dpower" % i] = tx_power['OpticalPowerTx%dField' %i]
+            bulk_status["tx%dbias" % i] = tx_bias['LaserBiasTx%dField' % i] if self.get_tx_bias_support() else 'N/A'
+            bulk_status["rx%dpower" % i] = rx_power['OpticalPowerRx%dField' %i] if self.get_tx_power_support() else 'N/A'
+            bulk_status["tx%dpower" % i] = tx_power['OpticalPowerTx%dField' %i] if self.get_rx_power_support() else 'N/A'
```

Comment:
please modify the definition of get_tx_power_support() and get_rx_power_support() as mentioned in above comment for get_tx_bias_support() API. Also, rx and tx APIs are swapped above.

Reply:
if you see the definition of  get_tx_power(), it should return N/A if its not supported, so there is no need to put checks in line 191, 192. so fix the api get_tx_power() and it should automatically fix here.

Reply:
modified the support function. For line 191 and 192, I am calling keys in the dictionary but they are non-existent if it returns N/A